### DISCUSSION TITLE
Configuring the Content Security Policy?

### DISCUSSION THREAD

**Discussion by rickcecil** [gildas-lormeau/SingleFile#926] [Q&A] [answered]
Hi! Loving SingleFile. I use it in a few different ways. Primarily, I use it with ArchiveBox. The issue is that I am using a bookmarklet that loads an annotator from a separate internal domain. The files that the SingleFile binary used on ArchiveBox include the meta tag with a content-security-policy declaration that prevents this bookmarklet from loading. 

Is there a way to configure the meta tag declaration that defines the content-security-policy? Any configuration options or other method? I'd even be willing to setup a custom binary for it if needed -- though would appreciate if you could point me in the right direction. Thanks!

**Comment by gildas-lormeau** [comment, marked as answer]
I just added the option `--insert-meta-csp`. You have to set it to `false` to ensure SingleFile won't add the `<meta>` tag. However, I don't know if you can pass your own options to SingleFile in ArchiveBox.

**Reply by rickcecil** [reply]
Awesome! Very much appreciated. :)

Did not expect such a fast turnaround. I think I can point to a custom binary for SingleFile in Archivebox.

**Reply by rickcecil** [reply]
(I hope this is okay to add. I've been trying to make this work and FINALLY got it to. Thought I would pass along what I discovered in case anyone stumbles across this thread and wonders how to do make this work.)

To anyone that finds this thread and wonders if you can add options for SingleFile within ArchiveBox. The answer is not yet. But it is coming. The code has been accepted. It does not appear to be showing up in the dev version (0.6.3) just yet. 

https://github.com/renaisun/ArchiveBox/commit/8899fe0b9259748da2ef19d37028c317f39f37d3

**Reply by gildas-lormeau** [reply]
Awesome! Thank you :)
